persistent_connection.go

```diff
@@ -177,6 +180,17 @@ func (pconn *persistentConn) readResponse() (int, string, error) {
 			err:       fmt.Errorf("error reading response: %s", err),
 			temporary: true,
 		}
+	} else if code == replyConnectionClosed {
+		pconn.controlConn.SetReadDeadline(time.Now().Add(pconn.config.Timeout))
+		code, msg, err = pconn.reader.ReadResponse(0)
```

Comment:
This was hard to reproduce as it happened randomly on a high latency server (probably because of some packet loss...). Looks like a bug in the server as it should not send a 426 code as a preliminary reply. In my tests it happened while walking with ReadDir() . 

It could be handled in dataStringList & transferFromOffset if I had more data but this seems like a more generic fix.
I have found no way of checking the socket buffer for data in go in a non-blocking manner (timeout is not a real option as it does not handle sub-second timeouts in my tests). Also does bufio Reader treat the connection break as EOF or returns an error?